paymentsheet/src/main/java/com/stripe/android/customersheet/data/CustomerSessionElementsSessionManager.kt

```diff
@@ -1,61 +1,61 @@
 package com.stripe.android.customersheet.data
 
+import com.stripe.android.common.validation.CustomerSessionClientSecretValidator
 import com.stripe.android.core.injection.IOContext
 import com.stripe.android.customersheet.CustomerSheet
 import com.stripe.android.customersheet.ExperimentalCustomerSheetApi
 import com.stripe.android.model.ElementsSession
+import com.stripe.android.payments.core.analytics.ErrorReporter
 import com.stripe.android.paymentsheet.ExperimentalCustomerSessionApi
 import com.stripe.android.paymentsheet.PaymentSheet
 import com.stripe.android.paymentsheet.PrefsRepository
 import com.stripe.android.paymentsheet.model.SavedSelection
 import com.stripe.android.paymentsheet.repositories.ElementsSessionRepository
 import kotlinx.coroutines.withContext
+import java.lang.IllegalArgumentException
 import javax.inject.Inject
 import javax.inject.Singleton
 import kotlin.coroutines.CoroutineContext
 
 internal interface CustomerSessionElementsSessionManager {
-    suspend fun fetchCustomerSessionEphemeralKey(): Result<CachedCustomerEphemeralKey.Available>
+    suspend fun fetchCustomerSessionEphemeralKey(): Result<CachedCustomerEphemeralKey>
 
-    suspend fun fetchElementsSession(): Result<ElementsSession>
+    suspend fun fetchElementsSession(): Result<CustomerSessionElementsSession>
```

Comment:
fwiw I think that `CustomerSessionElementsSession` could also be a completely internal type, and the return type of this function could be unchanged. I don't think there are any consumers of this function yet though, so it's unclear to me which will be more ergonomic!

Reply:
Actually `PaymentMethodDataSource` consumes the result of this function. The `type` helps ensure `ElementsSession.Customer` is not null unlike the `ElementsSession` type which can have it null. This way, `CustomerSessionElementsSessionManager` is responsible for ensuring that the `customer` field is available and not the individual data sources.

Reply:
oh gotcha. Nice! That's great